### DISCUSSION TITLE
Bugs and Features

### DISCUSSION THREAD

**Discussion by Stratecadia** [mrhan1993/Fooocus-API#380] [Ideas]
Hello mrhan1993, 

Firstly, kudos and well done with your work to date on this project! 

I am trying to work out how to manage files created and stored on the API server. It is important for managing disk resources that files can be deleted after being downloaded. I have found a function in file_utils.py called "delete_output_file(filename: str)", but it doesn't seem to ever get called. I propose that a route is created to delete the file specified by a given job number, so that can be called after the webhook endpoint completes the download.

Secondly, I have noted that if you download a file in base64 (async_process: False), the save_extension parameter does not work. For example if I set the save_extension to jpg, the file is always in a .png format, despite  the url parameter in the return object having a .jpg suffix.

Please let me know your thoughts, especially on the file delete function.

Cheers,

**Comment by mrhan1993** [comment]
Little problem. I will solve these later. 

by the way, Recommend my refactored version, even though it doesn't seem like everyone likes the new project very much :disappointed: https://github.com/mrhan1993/FooocusAPI

**Comment by mrhan1993** [comment]
Regarding deletion, there is only limited support, because if `--persistent` is not enabled, the program does not know where to delete files. Although there will be a memory queue to save history when the program is always open, this queue disappears after restart.

**Comment by mrhan1993** [comment]
a new branch for test [feature/delete_by_id](https://github.com/mrhan1993/Fooocus-API/tree/feature/delete_by_id)

maybe you can try it and feedback

**Comment by epixia** [comment]
Amazing work was wondering why this is no GUI interface form page to send the send prompt to API?
